src/utils/api.js

```diff
@@ -35,24 +35,36 @@ class Api {
 
 		this._refreshPromise = fetch(`${this.config.AUTH_SERVER}/api/v2/refresh_token`, {
 			method: 'POST',
-			credentials: 'include',
-		}).finally(() => { this._refreshPromise = null; });
+			credentials: 'omit',
+		}).then((res) => {
+			if (res.status > 400) {
+				throw res.json();
+			}
+			return res.json();
+		}).then(response => setAllLoginCookies({
+			accessToken: response.access_token,
+			refreshToken: response.refresh_token,
+			csrfToken: response.csrf_token,
+			userId: response.user_id,
+		})).finally(() => { this._refreshPromise = null; });
 
 		return this._refreshPromise;
 	}
 
-	_sendReq(method, path, body) {
-		return this._getCsrfCookie()
-			.then(cookie => fetch(`${this.config.ACCOUNT_SERVER}${path}`, {
-				method,
-				headers: {
-					'Content-Type': Api.JSONAPI_CONTENT_TYPE,
-					'Content-Length': Buffer.byteLength(JSON.stringify(body)),
-					'X-CSRF-Token': cookie,
-				},
-				body: JSON.stringify(body),
-				credentials: 'include',
-			}));
+	async _sendReq(method, path, body) {
+		const	accessTokenCookie = await cookiesGet({ name: 'access_token' });
+		const	csrfTokenCookie = await cookiesGet({ name: 'csrf_token' });
+		return fetch(`${this.config.ACCOUNT_SERVER}${path}`, {
+			method,
+			headers: {
+				'Content-Type': Api.JSONAPI_CONTENT_TYPE,
+				'Content-Length': Buffer.byteLength(JSON.stringify(body)),
+				Authorization: `Bearer ${accessTokenCookie ? accessTokenCookie.value : ''}`,
```

Comment:
Wouldn't it be better to leave skip the Authorization header if there is no access token? From what I understand, it is standardized to be
```
Authorization: <auth-scheme> <authorization-parameters>
```

```
Authorization: Bearer
```
is telling I'm authorized and my (base64-decoded) token is empty